### PR TITLE
BAH-2688 | Short name should be displayed for test and panels in Lab Lite.

### PR DESCRIPTION
**Pull request by abinaya-u**
## Requirements

- [x] This PR has a proper title that briefly describes the work done
- [x] I have squashed / amended the comments to make it more redable
- [x] I have included link to all the JIRA ticket('s)
- [x] I wrote the code as a pair or atleast performed a self-review of my own code
- [x] I have updated the documentation on [Bahmni Wiki](https://bahmni.atlassian.net/wiki/spaces/BAH/overview) or [README](https://github.com/Bahmni/bahmni-lab-frontend/blob/main/README.md) (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Summary
Short name should be displayed for test and panels in Lab Lite.

## JIRA tickets
https://bahmni.atlassian.net/browse/BAH-2688?atlOrigin=eyJpIjoiZDY2NGY5MTQzODY0NDVkYTljOGM1ZDBlZWM3MmVkNDciLCJwIjoiaiJ9

### CODE REVIEW

File: src/utils/helperFunctions.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import {LabTest} from '../types/selectTest'
```
Comment (gsluthra) [inline src/utils/helperFunctions.ts:1]: Why is this code in a helperFunctions() method. Shouldn't it either be in LabTest object, or a Panel object, or some class that represents LabTests? (Domain object class). This code seems to be outside any domain, as just "helper". 

Thoughts?

Comment (abinaya-u) [reply src/utils/helperFunctions.ts:1]: @gsluthra LabTest here is a data type which we are specifying in the function argument. The reason behind having getShortName() and getTestName() in a separate helperFunctions is so that these functions can be used across lablite where ever labtest names need to be displayed

File: src/context/lab-test-results-context.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import React, {useState} from 'react'
+import {LabTest} from '../types/selectTest'
+
+interface LabTestResultsContextProps {
+  labTestResults: any
```
Comment (kk-tw) [inline src/context/lab-test-results-context.tsx]: Can we avoid using any type. Will we be able to create a type and leverage the type

File: src/patient-lab-dashboard/patient-lab-details/patient-lab-details.test.tsx [hunk 3]
```diff
@@ -30,20 +30,18 @@ import {
...
-  mockReportTableResponse,
+  mockReportTableResponse
 } from '../../__mocks__/reportTable.mock'
 import {
-  diagnosticReportRequestBodyWithBasedOn,
-  mockDiagnosticReportErrorResponse,
-  mockDiagnosticReportResponse,
-  mockLabTestsResponse,
-  mockUploadFileResponse,
+  diagnosticReportRequestBodyWithBasedOn, mockDiagnosticReportErrorResponse,
```
Comment (kk-tw) [inline src/patient-lab-dashboard/patient-lab-details/patient-lab-details.test.tsx]: There is no need to reformat this. It is easier to read and modify to have one import per line

File: src/patient-lab-dashboard/patient-lab-details/patient-lab-details.tsx [hunk 4]
```diff
@@ -183,36 +184,39 @@ const PatientLabDetails: React.FC<RouteComponentProps<PatientParamsType>> = ({
...
+                    saveHandler={(isSaveSuccess = false) =>
+                      handleUploadAndSave(isSaveSuccess)
+                    }
+                    closeHandler={() => handleClose()}
+                    header="Upload Report"
+                    patientUuid={patientUuid}
+                  />
+                </UploadReportProvider>
+              )}
+              <br></br>
```
Comment (kk-tw) [inline src/patient-lab-dashboard/patient-lab-details/patient-lab-details.tsx]: It is generally advisable to having styling(margins) instead of adding line breaks. if there is a way to add space via margin please take that route

File: src/patient-lab-dashboard/patient-lab-details/patient-lab-details.test.tsx [hunk 5]
```diff
@@ -2,7 +2,7 @@ import {
   ExtensionSlot,
   openmrsFetch,
   usePagination,
-  usePatient,
+  usePatient
```
Comment (kk-tw) [inline src/patient-lab-dashboard/patient-lab-details/patient-lab-details.test.tsx]: The dangling comma can be retained

File: src/patient-lab-dashboard/table/report-table/report-table.component.tsx [hunk 6]
```diff
@@ -75,6 +97,25 @@ const ReportTable = (props: ReportTableProps) => {
     return selfPatient
   }
 
+  useEffect(() => {
+    if (testResults) {
+      setLabTestResults({...testResults})
+      const labOrder = testResults?.data?.results[0]
+      labOrder &&
+        getTestsInLabOrder(labOrder)?.map(sample => {
```
Comment (kk-tw) [inline src/patient-lab-dashboard/table/report-table/report-table.component.tsx]: I don't think map is being used correctly here, if the purpose is to loop through, please use forEach or a for loop

File: src/patient-lab-dashboard/table/report-table/report-table.component.tsx [hunk 7]
```diff
@@ -75,6 +97,25 @@ const ReportTable = (props: ReportTableProps) => {
...
+          getTestsInLabOrder(sample).map((tests: LabTest) => {
```
Comment (kk-tw) [inline src/patient-lab-dashboard/table/report-table/report-table.component.tsx]: map creates a new array and  it is advisable not to use map unless we need to create a new array

File: src/patient-lab-dashboard/table/report-table/report-table.component.tsx [hunk 8]
```diff
@@ -91,7 +132,10 @@ const ReportTable = (props: ReportTableProps) => {
 
         return {
           id: row.resource.id,
-          tests: row.resource.code.coding[0].display,
+          tests: getShortName(
+            row.resource.code.coding[0].display,
```
Comment (kk-tw) [inline src/patient-lab-dashboard/table/report-table/report-table.component.tsx]: validate if coding[0] is defined

File: src/patient-lab-dashboard/table/report-table/report-table.component.tsx [hunk 9]
```diff
@@ -114,135 +158,140 @@ const ReportTable = (props: ReportTableProps) => {
...
+                className={classes.reportTable}
+                useZebraStyles={true}
+              >
+                <TableHead>
+                  <TableRow>
+                    <TableExpandHeader id="expand" />
+                    {headers.map((header, i) => (
+                      <TableHeader
+                        id={header.header}
+                        key={i}
```
Comment (kk-tw) [inline src/patient-lab-dashboard/table/report-table/report-table.component.tsx]: please avoid using indices as keys. Attach a string so that it will be unique across the application

File: src/patient-lab-dashboard/table/report-table/report-table.component.tsx [hunk 10]
```diff
@@ -114,135 +158,140 @@ const ReportTable = (props: ReportTableProps) => {
...
+            pageNumber={currentPage}
+            pageSize={defaultPageSize}
+            currentItems={paginatedReportsTable?.length}
+            totalItems={rows?.length}
+            onPageNumberChange={({page}) => {
+              goTo(page)
+            }}
+          />
+        )}
+      </>
```
Comment (kk-tw) [inline src/patient-lab-dashboard/table/report-table/report-table.component.tsx:293]: Suggestion: This entire JSX is very hard to follow. We can split it into smaller components for easier maintenance

File: src/context/lab-test-results-context.tsx (hunk 2 above)
Comment (abinaya-u) [reply src/context/lab-test-results-context.tsx]: changes have been made, introduced new type

File: src/patient-lab-dashboard/patient-lab-details/patient-lab-details.test.tsx (hunk 3 above)
Comment (abinaya-u) [reply src/patient-lab-dashboard/patient-lab-details/patient-lab-details.test.tsx]: changes have been made

File: src/patient-lab-dashboard/patient-lab-details/patient-lab-details.tsx (hunk 4 above)
Comment (abinaya-u) [reply src/patient-lab-dashboard/patient-lab-details/patient-lab-details.tsx]: added css instead of break points

File: src/patient-lab-dashboard/patient-lab-details/patient-lab-details.test.tsx (hunk 5 above)
Comment (abinaya-u) [reply src/patient-lab-dashboard/patient-lab-details/patient-lab-details.test.tsx]: changes have been made

File: src/patient-lab-dashboard/table/report-table/report-table.component.tsx (hunk 6 above)
Comment (abinaya-u) [reply src/patient-lab-dashboard/table/report-table/report-table.component.tsx]: introduced foreach instead of map

File: src/patient-lab-dashboard/table/report-table/report-table.component.tsx (hunk 8 above)
Comment (abinaya-u) [reply src/patient-lab-dashboard/table/report-table/report-table.component.tsx]: changes have been made

File: src/patient-lab-dashboard/table/report-table/report-table.component.tsx (hunk 9 above)
Comment (abinaya-u) [reply src/patient-lab-dashboard/table/report-table/report-table.component.tsx]: changes have been made

File: src/patient-lab-dashboard/table/report-table/report-table.component.tsx (hunk 10 above)
Comment (abinaya-u) [reply src/patient-lab-dashboard/table/report-table/report-table.component.tsx:293]: so the entire return is a DataTable in carbon component, thats why didn't split it into smaller components. Let me know your thoughts

Comment (kk-tw) [reply src/patient-lab-dashboard/table/report-table/report-table.component.tsx:293]: This is alright for now. We can extract the TableBody though. You can retain it for now.

File: src/patient-lab-dashboard/select-test/select-test.tsx [hunk 11]
```diff
@@ -45,24 +43,20 @@ const SelectTest = ({isDiscardButtonClicked}) => {
...
-      labTestResults &&
-      getTestsInLabOrder(labOrder)?.map(sample => {
-        getTestsInLabOrder(sample).map((tests: LabTest) => {
-          if (isLabTest(tests))
-            setTotalTests(totalTest => [...totalTest, tests])
-          setSearchResults(searchResults => [...searchResults, tests])
-          setAllTestsAndPanels(searchResults => [...searchResults, tests])
-        })
+    allTestsAndPanels &&
+      allTestsAndPanels.map((tests: LabTest) => {
```
Comment (kk-tw) [inline src/patient-lab-dashboard/select-test/select-test.tsx]: Missed this here. This map can also be replaced with a forEach

**Review by kk-tw — APPROVED** [review]

File: src/patient-lab-dashboard/select-test/select-test.tsx (hunk 11 above)
Comment (abinaya-u) [reply src/patient-lab-dashboard/select-test/select-test.tsx]: fixed
